install/kubernetes/cilium/templates/cilium-agent/service.yaml

```diff
@@ -44,3 +45,24 @@ spec:
     targetPort: envoy-metrics
 {{- end }}
 {{- end }}
+{{- if and .Values.hubble.enabled .Values.hubble.listenAddress .Values.hubble.peerService.enabled }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: hubble-peer
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: cilium
+spec:
+  selector:
+    k8s-app: cilium
+  ports:
+  - name: peer-service
+    port: {{ .Values.hubble.peerService.servicePort }}
```

Comment:
Could we have the same logic for the port as for the Hubble Relay service (443 when TLS is enabled, 80 otherwise unless the port is user specified). Below is how it's done for the Relay service:
```  ports:
  - protocol: TCP
  {{- if .Values.hubble.relay.servicePort }}
    port: {{ .Values.hubble.relay.servicePort }}
  {{- else }}
    port: {{ .Values.hubble.relay.tls.server.enabled | ternary 443 80 }}
  {{- end }}
    targetPort: {{ .Values.hubble.relay.listenPort }}
```

Reply:
Can I ask why? The peer service is a mostly internal service, there is no reason to make the service more legible to cluster operators.

Reply:
If there's a service, then I don't think it can be considered "internal" only anymore anyways. Note that the hubble CLI already supports querying the service (the command is hidden but try `hubble watch peer`).